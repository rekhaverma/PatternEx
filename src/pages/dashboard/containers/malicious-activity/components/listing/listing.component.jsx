import React from 'react';
import PropTypes from 'prop-types';

import MultiSearch from 'components/multi-search';
import Table from 'components/table-v2';
import Loader from 'components/loader/loader-v2.component';
import { EvpOpenMethods } from 'model/classes/evp-open-methods';

import { maliciousColumns } from '../../constants';
import { filterByMultipleTags } from '../../utils';

export class Listing extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      tags: [],
    };
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onSearchChange(tags) {
    this.setState({ tags });
  }

  getFilteredData() {
    const {
      data,
      isOldEVPActive,
      handleExplodedView,
      setLabelForPrediction,
      deleteLabel,
    } = this.props;

    const filteredData = filterByMultipleTags(data, this.state.tags);

    return filteredData.map(el => ({
      ...el,
      'handlers': {
        ...el.handlers,
        'onRowClick': (e, row) => EvpOpenMethods.onRowClickHandler(row, 'malicious', isOldEVPActive, handleExplodedView),
        'onInspect': (e, row) => EvpOpenMethods.onInspectHandler(row, 'malicious', isOldEVPActive),
        'getNewTabUrl': row => EvpOpenMethods.getNewTabUrlHandler(row, 'malicious', isOldEVPActive),
        'onConfirm': (e) => {
          e.stopPropagation();
          const params = {
            ...el,
            start_time: el.end_time,
          };
          setLabelForPrediction(params, true);
        },
        'onDeny': (e) => {
          e.stopPropagation();
          setLabelForPrediction(el, false);
        },
        'deleteLabel': (e) => {
          e.stopPropagation();
          if (el.user_tag && el.user_tag.label_id) {
            deleteLabel(el.user_tag.label_id);
          }
        },
      },
    }));
  }

  render() {
    const {
      isDataLoaded,
      searchTags,
    } = this.props;

    if (!isDataLoaded) {
      return (
        <div className="dashboard__column">
          <Loader small />
        </div>
      );
    }

    const filteredData = this.getFilteredData();
    const tags = searchTags.concat(this.state.tags);

    return (
      <div className="dashboard__column full-width no-padding">
        <div className="dashboard__topColumn" style={{ display: 'flex' }}>
          <MultiSearch
            tags={tags}
            onChange={this.onSearchChange}
            placeholder="Search for entity or threat tactic"
          />
        </div>
        <Table
          classname="table"
          data={filteredData}
          tableConfig={maliciousColumns}
          pageSize="10"
        />
      </div>
    );
  }
}

Listing.propTypes = {
  isDataLoaded: PropTypes.bool.isRequired,
  data: PropTypes.array.isRequired,
  handleExplodedView: PropTypes.func.isRequired,
  deleteLabel: PropTypes.func.isRequired,
  setLabelForPrediction: PropTypes.func.isRequired,
  isOldEVPActive: PropTypes.bool,
  searchTags: PropTypes.array,
};

Listing.defaultProps = {
  searchTags: [],
  isOldEVPActive: false,
};
