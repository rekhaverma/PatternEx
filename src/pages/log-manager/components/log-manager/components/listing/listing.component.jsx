import React from 'react';
import PropTypes from 'prop-types';

import { SmartTable } from 'components/smart-table';

import { tableConfig } from '../../constants';

const Listing = (props) => {
  const { className, data } = props;
  return (
    <div className={`${className}__listing`}>
      <SmartTable
        tableId="log-manager/log-manager"
        hasMultiTagSearch
        data={data}
        tableConfig={tableConfig}
      />
    </div>
  );
};

Listing.propTypes = {
  data: PropTypes.array.isRequired,
  className: PropTypes.string,
};

Listing.defaultProps = {
  className: 'log-manager',
};

export default Listing;
