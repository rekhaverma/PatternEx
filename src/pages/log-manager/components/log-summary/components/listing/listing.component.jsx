import React from 'react';
import PropTypes from 'prop-types';

import { SmartTable } from 'components/smart-table';

import { tableConfig } from '../../constants';
import { ExpandedRow } from '../expanded-row/expanded-row.component';

export const Listing = (props) => {
  const { className, data } = props;
  const dataWithExpandData = data.map(item => ({
    ...item,
    expandedData: (
      <ExpandedRow
        data={Object.keys(item.logSummaryEntities).map(key => ({
          title: key,
          value: item.logSummaryEntities[key],
        }))}
        className={className}
      />
    ),
  }));
  return (
    <div className={`${className}__listing`}>
      <SmartTable
        tableId="log-manager/log-summary"
        hasMultiTagSearch
        acceptExpand
        expandedHeight={155}
        data={dataWithExpandData}
        tableConfig={tableConfig}
      />
    </div>
  );
};

Listing.propTypes = {
  data: PropTypes.array.isRequired,
  className: PropTypes.string,
};

Listing.defaultProps = {
  className: 'log-manager',
};
