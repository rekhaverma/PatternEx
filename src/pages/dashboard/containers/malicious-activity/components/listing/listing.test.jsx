import React from 'react';
import moment from 'moment';
import { Listing } from './listing.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<Listing />', () => {
  it('should match with snapshot - with loading', () => {
    const props = {
      isDataLoaded: false,
      isOldEVPActive: true,
      data: [],
      handleExplodedView: jest.fn(),
      goBackUrl: jest.fn(),
      deleteLabel: jest.fn(),
      setLabelForPrediction: jest.fn(),
    };
    const wrapper = shallow(<Listing {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - w/o data', () => {
    const props = {
      isDataLoaded: true,
      data: [],
      handleExplodedView: jest.fn(),
      goBackUrl: jest.fn(),
      deleteLabel: jest.fn(),
      setLabelForPrediction: jest.fn(),
      time: {
        startTime: moment.utc('2018-10-11'),
        endTime: moment.utc('2018-10-11'),
        timezone: 'UTC',
      },
    };
    const wrapper = shallow(<Listing {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - with data', () => {
    const props = {
      isDataLoaded: true,
      data: [{}],
      handleExplodedView: jest.fn(),
      goBackUrl: jest.fn(),
      deleteLabel: jest.fn(),
      setLabelForPrediction: jest.fn(),
      time: {
        startTime: moment.utc('2018-10-11'),
        endTime: moment.utc('2018-10-11'),
        timezone: 'UTC',
      },
    };
    const wrapper = shallow(<Listing {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
