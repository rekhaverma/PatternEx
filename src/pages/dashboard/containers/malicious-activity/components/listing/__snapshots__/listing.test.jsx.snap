// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Listing /> should match with snapshot - w/o data 1`] = `
<div
  className="dashboard__column full-width no-padding"
>
  <div
    className="dashboard__topColumn"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <MultiSearch
      className="multi-search"
      onChange={[Function]}
      placeholder="Search for entity or threat tactic"
      style={Object {}}
      tags={Array []}
    />
  </div>
  <ConfigurableTable
    classname="table"
    data={Array []}
    maxPageSize="100"
    onRowClick={[Function]}
    pageSize="10"
    rowHeight={30}
    tableConfig={
      Array [
        Object {
          "className": "table__headerColumn",
          "field": "hash",
          "hidden": true,
          "intl": "malicious.table.hash",
          "isKey": true,
          "label": "Hash",
          "resizable": false,
        },
        Object {
          "className": "table__headerColumn",
          "field": "start_time_formatted",
          "hidden": false,
          "intl": "malicious.table.start_time_formatted",
          "isKey": false,
          "label": "Log Date",
          "resizable": true,
          "width": 100,
        },
        Object {
          "className": "table__headerColumn--entity",
          "field": "entity_name",
          "hidden": false,
          "intl": "malicious.table.entity_name",
          "isKey": false,
          "label": "Entity",
          "resizable": true,
          "tooltip": "investigate",
        },
        Object {
          "className": "table__headerColumn",
          "field": "score",
          "hidden": false,
          "intl": "malicious.table.score",
          "isKey": false,
          "label": "Score",
          "resizable": true,
          "width": 150,
        },
        Object {
          "className": "table__headerColumn--highlighted",
          "field": "threat",
          "hidden": false,
          "intl": "malicious.table.threat",
          "isKey": false,
          "label": "Threat Tactic",
          "resizable": true,
          "width": 220,
        },
        Object {
          "className": "table__headerColumn--actions",
          "field": "actions",
          "hidden": false,
          "intl": "malicious.table.actions",
          "isKey": false,
          "label": "Actions",
          "resizable": true,
          "sort": false,
          "width": 250,
        },
      ]
    }
  />
</div>
`;

exports[`<Listing /> should match with snapshot - with data 1`] = `
<div
  className="dashboard__column full-width no-padding"
>
  <div
    className="dashboard__topColumn"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <MultiSearch
      className="multi-search"
      onChange={[Function]}
      placeholder="Search for entity or threat tactic"
      style={Object {}}
      tags={Array []}
    />
  </div>
  <ConfigurableTable
    classname="table"
    data={
      Array [
        Object {
          "handlers": Object {
            "deleteLabel": [Function],
            "getNewTabUrl": [Function],
            "onConfirm": [Function],
            "onDeny": [Function],
            "onInspect": [Function],
            "onRowClick": [Function],
          },
        },
      ]
    }
    maxPageSize="100"
    onRowClick={[Function]}
    pageSize="10"
    rowHeight={30}
    tableConfig={
      Array [
        Object {
          "className": "table__headerColumn",
          "field": "hash",
          "hidden": true,
          "intl": "malicious.table.hash",
          "isKey": true,
          "label": "Hash",
          "resizable": false,
        },
        Object {
          "className": "table__headerColumn",
          "field": "start_time_formatted",
          "hidden": false,
          "intl": "malicious.table.start_time_formatted",
          "isKey": false,
          "label": "Log Date",
          "resizable": true,
          "width": 100,
        },
        Object {
          "className": "table__headerColumn--entity",
          "field": "entity_name",
          "hidden": false,
          "intl": "malicious.table.entity_name",
          "isKey": false,
          "label": "Entity",
          "resizable": true,
          "tooltip": "investigate",
        },
        Object {
          "className": "table__headerColumn",
          "field": "score",
          "hidden": false,
          "intl": "malicious.table.score",
          "isKey": false,
          "label": "Score",
          "resizable": true,
          "width": 150,
        },
        Object {
          "className": "table__headerColumn--highlighted",
          "field": "threat",
          "hidden": false,
          "intl": "malicious.table.threat",
          "isKey": false,
          "label": "Threat Tactic",
          "resizable": true,
          "width": 220,
        },
        Object {
          "className": "table__headerColumn--actions",
          "field": "actions",
          "hidden": false,
          "intl": "malicious.table.actions",
          "isKey": false,
          "label": "Actions",
          "resizable": true,
          "sort": false,
          "width": 250,
        },
      ]
    }
  />
</div>
`;

exports[`<Listing /> should match with snapshot - with loading 1`] = `
<div
  className="dashboard__column"
>
  <Loader
    small={true}
  />
</div>
`;
